File: HarvardX-CS50-project02-Mario(more)/C/mario-more.c
#include <cs50.h>
#include <stdio.h>

void print_symbol(char symbol, int right_margin);
int main(void)
{
    int height;
    char symbol = '#';
    char offset_symbol = ' ';
    string separator = "  ";
    
    do
    {
        height = get_int("Height: ");
    }
    while(height < 1 || height > 8);
    
     for(int i = 0; i < height; i++)
    {
        for(int m = height - 1; m > i; m--)
        {
            printf("%c", offset_symbol);
        }
        print_symbol(symbol, i);
         
        printf("%s", separator);
         
        print_symbol(symbol, i);
         
        printf("\n");
        
    }
    
   
    
}

void print_symbol(char symbol, int right_margin)
{
    for(int k = 0; k < (right_margin+1); k++)
        {
            printf("%c", symbol);
        }
}

File: HarvardX-CS50-project02-Mario(more)/Python/mario-more.py
def main():
    height = 0
    symbol = '#'
    offset_symbol = ' '
    separator = "  "

# validate input
    while True:
        height = input("Height: ")

        if height.isdigit():
            height = int(height)

            if height > 0 and height < 9:
                break

    # print pyramid
    for i in range(height):
        for j in range(height - 1, i, -1):
            print(f"{offset_symbol}", end="")

        print_symbol(symbol, i)
        print(f"{separator}", end="")
        print_symbol(symbol, i)
        print()

# functions declarations
def print_symbol(symbol, right_margin):
    for i in range(right_margin + 1):
        print(f"{symbol}", end="")

if __name__ == "__main__":
    main()
